pittAPI.py

```diff
@@ -495,7 +495,10 @@ def get_dining_locations_by_status(self, status=None):
         return dining_locations
 
     def get_dining_location_by_name(self, location):
-        return self.get_dining_locations()[self._encode_dining_location(location)]
+        try:
+            return self.get_dining_locations()[self._encode_dining_location(location)]
+        except:
+            return {}
```

Comment:
I would throw an error. Silent failures aren't fun and Exceptions exist to inform the programmer they messed up somewhere. Sorry for the long review! This is some really high quality work, there's just a lot of it.